src/EditorFeatures/CSharpTest/MoveToNamespace/MoveToNamespaceTests.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright(c) Microsoft.All Rights Reserved.Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.MoveToNamespace;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities.MoveToNamespace;
+using Microsoft.VisualStudio.Composition;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MoveToNamespace
+{
+    public class MoveToNamespaceTests : AbstractCSharpCodeActionTest
+    {
+        private static readonly IExportProviderFactory CSharpExportProviderFactory =
+            ExportProviderCache.GetOrCreateExportProviderFactory(
+                TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic
+                    .WithPart(typeof(TestMoveToNamespaceOptionsService)));
+
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new MoveToNamespaceCodeActionProvider();
+
+        protected override TestWorkspace CreateWorkspaceFromFile(string initialMarkup, TestParameters parameters)
+            => TestWorkspace.CreateCSharp(initialMarkup, parameters.parseOptions, parameters.compilationOptions, exportProvider: CSharpExportProviderFactory.CreateExportProvider());
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.MoveToNamespace)]
+        public Task MoveToNamespace_MoveItems_CaretOnNamespaceName()
+            => TestInRegularAndScriptAsync(
+@"namespace A[||] 
+{
+    class MyClass
+    {
+        void Method() { }
+    }
+}",
+@$"namespace {{|Warning:{TestMoveToNamespaceOptionsService.NamespaceValue}|}}
```

Comment:
What's the "Warning" thing about in these tests?

Reply:
Changing namespaces can result in invalid code (or some text to that effect). It's a general warning that gets applied to refactorings. Moving to the full CodeAction testing infra resulted in these being added. The user won't see these, but the tests verify that they are expected; in this case, they are.